src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -97,140 +140,93 @@ void Terminal::MultiClickSelection(const COORD viewportPos, SelectionExpansionMo
 // - position: the (x,y) coordinate on the visible viewport
 void Terminal::SetSelectionAnchor(const COORD viewportPos)
 {
-    _selectionPivot = _ConvertToBufferCell(viewportPos);
-    _buffer->GetSize().Clamp(_selectionPivot);
+    _selection = SelectionAnchors{};
+    _selection->pivot = _ConvertToBufferCell(viewportPos);
 
-    _multiClickSelectionMode = SelectionExpansionMode::Cell;
-
-    // Unlike MultiClickSelection(), we need to set
-    // these vars BEFORE SetSelectionEnd()
     _allowSingleCharSelection = (_copyOnSelect) ? false : true;
-    _selectionStart = _selectionPivot;
 
+    _multiClickSelectionMode = SelectionExpansionMode::Cell;
     SetSelectionEnd(viewportPos);
+
+    _selection->start = _selection->pivot;
 }
 
 // Method Description:
 // - Update selection anchors when dragging to a position
 // - based on the selection expansion mode
 // Arguments:
 // - viewportPos: the (x,y) coordinate on the visible viewport
-// - expansionModeOverwrite: overwrites the _multiClickSelectionMode for this function call. Used for ShiftClick
-void Terminal::SetSelectionEnd(const COORD viewportPos, std::optional<SelectionExpansionMode> expansionModeOverwrite)
+// - newExpansionMode: overwrites the _multiClickSelectionMode for this function call. Used for ShiftClick
+void Terminal::SetSelectionEnd(const COORD viewportPos, std::optional<SelectionExpansionMode> newExpansionMode)

```

Comment:
this function will throw if SetSelectionStart has not been called. Presume that's okay?